code/modules/food_and_drinks/kitchen_machinery/kitchen_machine.dm

```diff
@@ -125,37 +114,33 @@
 			add_item(O, user)
 	else if(is_type_in_list(O, list(/obj/item/reagent_containers/glass, /obj/item/reagent_containers/food/drinks, /obj/item/reagent_containers/food/condiment)))
 		if(!O.reagents)
-			return 1
+			return TRUE
 		for(var/datum/reagent/R in O.reagents.reagent_list)
 			if(!(R.id in GLOB.cooking_reagents[recipe_type]))
 				to_chat(user, "<span class='alert'>Your [O] contains components unsuitable for cookery.</span>")
-				return 1
-		//G.reagents.trans_to(src,G.amount_per_transfer_from_this)
+				return TRUE
 	else if(istype(O,/obj/item/grab))
 		return special_attack(O, user)
 	else
 		to_chat(user, "<span class='alert'>You have no idea what you can cook with [O].</span>")
-		return 1
-	updateUsrDialog()
+		return TRUE
+	SStgui.update_uis(src)
 
 /obj/machinery/kitchen_machine/proc/add_item(obj/item/I, mob/user)
 	if(!user.drop_item())
 		to_chat(user, "<span class='notice'>\The [I] is stuck to your hand, you cannot put it in [src]</span>")
-		//return 0
-	else
-		I.forceMove(src)
-		user.visible_message("<span class='notice'>[user] adds [I] to [src].</span>", "<span class='notice'>You add [I] to [src].</span>")
+		return
 
-/obj/machinery/kitchen_machine/attack_ai(mob/user)
-	return 0
+	I.forceMove(src)
+	user.visible_message("<span class='notice'>[user] adds [I] to [src].</span>", "<span class='notice'>You add [I] to [src].</span>")
+	SStgui.update_uis(src)
```

Comment:
can you check to see if you have your TGUI set to autoupdate, if it is (even if automatically) you don't need to call the update_uis proc

Reply:
Gonna be honest... I don't know what that is. Can you tell me?

Reply:
`SStgui.update_uis(src)` forces the UI to update with all the new data sent to it. Normally all TGUI by default will autoupdate unless otherwise specified (which I don't think is the case here).

This just means you can literally remove all of these